src/utils/test/packet_generators.rs

```diff
@@ -52,3 +52,47 @@ impl Stream for LinearIntervalGenerator {
         }
     }
 }
+
+/// Packet Interval Generator procduces a Stream of packets on a defined interval.AsMut
```

Comment:
The "LineraInterval" of `LinearIntervalGenerator` was meant to indicate that some packets are generated at a _fixed_ interval, e.g. every 100ms. The name `PacketIntervalGenerator` doesn't really make sense in this way. Maybe we should rename it to `FixedIntervalGenerator`, since you could also interpret the "Interval" as the ranges of digits, but that's not the intention. Let's deprecate `LinearIntervalGenerator`, since all ranges can be expressed by passing `range.into_iter()` to a `FixedIntervalGenerator`.

Reply:
We should also try to deprecate `LinearIntervalGenerator` since there's a lot of duplicated logic between the two.

Reply:
Leave deprecation for testing fixup issue.  I get the names are bad, but we shouldn't change too much in one PR